### PR TITLE
feat: add list comprehension tutorial

### LINKED ISSUE #130: [tutorial] list comprehension
**Description of the tutorial**
add list comprehension tutorial

**Do you want to be assigned to work on this?**
(mark [x])
- [x] YES
- [ ] NO

**Optional: Any additional info you want to add**
...

### PR DESCRIPTION
**Pull request by jrafaaael**
### 2. Description
Add list comprehension tutorial

### 3. Fill in checklist by marking [x]

- [x] I've read the CONTRIBUTING.md
- [x] I was assigned to this issue
- [x] My code is formatted using required linting
- [x] I've run my code locally and checked it works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Each file contains:
 |How to alter a list                        | [alterning-lists.py](lists/altering-lists.py)                  | 
 |How to declare a list                      | [declaring-lists.py](lists/declaring-lists.py)                 | 
 |How to index a list                        | [indexing-lists.py](lists/indexing-lists.py)                   | 
+|Comprehension                              | [list-comprehension.py](lists/comprehension.py)                |
```
Comment (sabderemane) [inline README.md:41]:
```suggestion
|List comprehension                              | [list-comprehension.py](lists/list-comprehension.py)                |
```
you should rename the file list-comprehension for consistency

File: lists/comprehension.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# In general there is a very common for loop structure that looks like this:
+for_loop_squares = []
+
+for i in range(10):
+    for_loop_squares.append(i * i)
+
+print(for_loop_squares)  # [0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
+
+
+# With list comprehension you can do the exact same thing:
```
Comment (sabderemane) [inline lists/comprehension.py:17]:
```suggestion
# With list comprehension you can do the exact same thing in one line.
# the syntax is : 
# new_list = [expression for member in iterable]
```

File: lists/comprehension.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+comprehension_squares = [x * x for x in range(10)]
+
+print(comprehension_squares)  # [0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
+
+
+# You can even add conditionals to the list comprehension:
```
Comment (sabderemane) [inline lists/comprehension.py:23]:
```suggestion
# You can even add conditionals to the list comprehension, the condition is place at the end of the list comprehension, just like that :
# new_list = [expression for member in iterable if condition]
```

File: lists/comprehension.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+comprehension_even_squares_numbers = [x * x for x in range(10) if x % 2 == 0]
+
+print(comprehension_even_squares_numbers)  # [0, 4, 16, 36, 64]
+
+
+# But if you can do if/else logic, you need to do:
```
Comment (sabderemane) [inline lists/comprehension.py:29]:
```suggestion
# But if you can do if/else logic, you need to place the condition before the for loop just like that:
```

File: lists/comprehension.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+comprehension_even_squares_numbers_and_odd_numbers_is_zero = [x * x if x % 2 == 0 else 0 for x in range(10)]
+
+print(comprehension_even_squares_numbers_and_odd_numbers_is_zero)  # [0, 0, 4, 0, 16, 0, 36, 0, 64, 0]
+
+
+# As you can see, the basic format of a list comprehension is like:
```
Comment (sabderemane) [inline lists/comprehension.py:35]:
```suggestion
# We can summarize the 3 formats of list comprehension. The basic format of a list comprehension is like:
```

File: lists/comprehension.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+# With if/else logic:
+# new_list = [expression if condition else other_expression for x in sequence]
+
+
+# ------------------------------------------------------------------------------------
+# Challenge: FIZZBUZZ
+# Create a list comprehesion that returns the numbers from 0 to 100 but:
+#   1 -. if the number is divisible by 3 and 5, append to the list the word "FizzBuzz"
+#   3 -. else, append to the list the number
```
Comment (sabderemane) [inline lists/comprehension.py:49]: Why there is "1 -" and 3 -"  ?

**Review by sabderemane — CHANGES_REQUESTED** [review]
Thank you @jrafaaael for your contribution 😄 I have suggested some changes to improve the tutorial

File: lists/comprehension.py (hunk 6 above)
Comment (jrafaaael) [reply lists/comprehension.py:52]: Oh, sorry! Originally I added three steps

**Comment by jrafaaael** [comment]
@sabderemane suggestions accepted! :smile: I think that the github-action job is trying to run the `.md` file :laughing:

**Review by sabderemane — APPROVED** [review]

**Comment by sabderemane** [comment]
Yeah, we have a bug for that 😅  I have checked manually, it's good for me, thank you @jrafaaael !

**Pull request merged**